activerecord/test/cases/relation/distinct_test.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
```

Comment:
I think there is no need to create a separate test class for this - can be placed in one of the existing ones.

Reply:
I couldn't find a single suite for query methods (nor any that are specifically for `#distinct`). And since there are classes like these:

- https://github.com/rails/rails/blob/main/activerecord/test/cases/relation/merging_test.rb
- https://github.com/rails/rails/blob/main/activerecord/test/cases/excluding_test.rb

I opted to create a separate one.

Reply:
I think it can be placed after https://github.com/rails/rails/blob/0f9aaa5ca9b8421ebf42b3d7720632d4c1cae5fc/activerecord/test/cases/relations_test.rb#L1782

And for `distinct.count`, there are lots of tests in `calculations_test.rb`. So I would left only the second test and made it simpler - no need to use associations/where/etc, a simple
```ruby
assert_raises(..., match: ...) do
  Post.distinct("not_true")
end
```
would be enough.

Reply:
I've moved the test to where you suggested and simplified it.